preprocess/file.go

```diff
@@ -0,0 +1,63 @@
+package preprocess
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	"github.com/pkg/errors"
+)
+
+const peekLen = 64 // look at first N bytes to figure out if it has shebang
+
+// textExtensions is predefined list of script and text
+// file extensions.
+var textExtensions = []string{
```

Comment:
they are mostly from the current version. I think I just added .txt and .pl. Can be extended but I'd say let's not if there's no clear need.